annotations.md

```diff
@@ -0,0 +1,228 @@
+## Binding Metadata in Annotations
+
+ During a binding operation, annotations from relevant Kubernetes resources are extracted to gather information about what is interesting for binding. This information is eventually used to bind the application with the backing service by populating the binding Secret.
+
+### Requirements for specifying binding information in a backing service CRD / Kubernetes resource
```

Comment:
Just trying to see if any of these scenarios can extract binding info from `Kafka` CR e.g.

```yaml
status:
  listeners:
    - type: plain
      addresses:
        - host: my-cluster-kafka-bootstrap.service-binding-demo.svc
          port: 9092
        - host: my-cluster-kafka-bootstrap.service-binding-demo.svc
          port: 9093
    - type: tls
      addresses:
        - host: my-cluster-kafka-bootstrap.service-binding-demo.svc
          port: 9094
```

--- 

I'm wondering if supporting templating would cover cases we haven't thought of. Maybe something like:

```yaml
    - path: listeners
      x-descriptors:
        - servicebinding:prefix:elementType=template:source:{{ GO_TEMPLATE }}:bindAs=volume
```

And the Go templates generates the following:
```
plain_0=my-cluster-kafka-bootstrap.service-binding-demo.svc:9092
plain_1=my-cluster-kafka-bootstrap.service-binding-demo.svc:9093
tls_0=my-cluster-kafka-bootstrap.service-binding-demo.svc:9094
```

After adding the prefix, the secret would look like this:
```
PREFIX_PLAIN_0: my-cluster-kafka-bootstrap.service-binding-demo.svc:9092
...
```

Reply:
Escape hatch! :) Let me think.

Reply:
ya - I think that's a great point @navidsh.    This essentially allows the Operator author to have these complex mappings without changing their controller code and without requiring each consumer to have to deal with that complexity in the `dataMappings` portion of their ServiceBinding.  

@sbose78 - I believe if we introduced Navid's proposal then we could close [this issue](https://github.com/redhat-developer/service-binding-operator/issues/300).

Reply:
@navidsh could you help me with a complete example please? I will work on introducing this in a separate PR today.